internal/acceptance/init_test.go

```diff
@@ -207,10 +242,16 @@ func run(t *testing.T, steps []step) {
 					if p[0] == "data" {
 						continue
 					}
-					r := provider.ResourcesMap[p[0]]
-					dia := r.ReadContext(ctx, r.Data(is.Primary), client)
-					if dia != nil {
-						return fmt.Errorf("%v", dia)
+
+					// Convert InstanceState from terraform-plugin-framework to terraform-plugin-sdk v2
+					sdkInstanceState := convertToSDKInstanceState(is.Primary)
+
+					if r, ok := sdkPluginProvider.ResourcesMap[p[0]]; ok {
+						// Only checking for sdkv2 resources.
+						dia := r.ReadContext(ctx, r.Data(sdkInstanceState), client)
+						if dia != nil {
+							return fmt.Errorf("%v", dia)
+						}
```

Comment:
I'm wondering if we really need this at all, even for SDK-based resources. If the test framework refreshes resources to read them again, then I think it is actually OK to remove this entirely. What do you think?

Reply:
I think maybe for now let's keep it just to minimize the amount of changes we introduce?